platform/ext/target/nordic_nrf/common/core/nrf_exception_info.c

```diff
@@ -11,34 +11,60 @@
 
 static struct nrf_exception_info nrf_exc_info;
 
-static void spu_dump_context(struct nrf_exception_info *ctx)
+static void dump_exception_info(struct nrf_exception_info *ctx)
 {
     SPMLOG_ERRMSG("Platform Exception: SPU Fault\r\n");
 
-#ifdef NRF_SPU
     /* Report which type of violation occured */
     if (ctx->events & SPU_EVENT_RAMACCERR) {
-        SPMLOG_DBGMSG("  RAMACCERR\r\n");
+        SPMLOG_DBGMSG("  SPU.RAMACCERR\r\n");
     }
 
     if (ctx->events & SPU_EVENT_PERIPHACCERR) {
-        SPMLOG_DBGMSG("  PERIPHACCERR\r\n");
+        SPMLOG_DBGMSG("  SPU.PERIPHACCERR\r\n");
+        SPMLOG_DBGMSGVAL(" Target addr: ", ctx->periphaccerr.address);
     }
 
     if (ctx->events & SPU_EVENT_FLASHACCERR) {
-        SPMLOG_DBGMSG("  FLASHACCERR\r\n");
+        SPMLOG_DBGMSG("  SPU.FLASHACCERR\r\n");
     }
-#else
-	// TODO: NCSDK-25011: Support error handling on nrf54l
-#endif
 
+#if MPC_PRESENT
+    if (ctx->events & MPC_EVENT_MEMACCERR) {
+        SPMLOG_DBGMSG("  MPC.MEMACCERR\r\n");
+        SPMLOG_DBGMSGVAL("  Target addr:          ", ctx->memaccerr.address);
+        SPMLOG_DBGMSGVAL("  Access information:   ", ctx->memaccerr.info);
+        SPMLOG_DBGMSGVAL("    Owner id:     ", ctx->memaccerr.info & 0xf);
+        SPMLOG_DBGMSGVAL("    Masterport:   ", (ctx->memaccerr.info & 0x1f0) >> 4);
+        SPMLOG_DBGMSGVAL("    Read:         ", (ctx->memaccerr.info >> 12) & 1);
+        SPMLOG_DBGMSGVAL("    Write:        ", (ctx->memaccerr.info >> 13) & 1);
+        SPMLOG_DBGMSGVAL("    Execute:      ", (ctx->memaccerr.info >> 14) & 1);
+        SPMLOG_DBGMSGVAL("    Secure:       ", (ctx->memaccerr.info >> 15) & 1);
+        SPMLOG_DBGMSGVAL("    Error source: ", (ctx->memaccerr.info >> 16) & 1);
+    }
+#endif
 }
 
 void nrf_exception_info_store_context(void)
 {
     nrf_exc_info.events = spu_events_get();
```

Comment:
Does that not need to be inside the ifdef?

Reply:
No, this is platform independent, the code in the ifdefs are to get extra information on supported platforms.